File: src/AiTrainer.Web/AiTrainer.Web.Domain.Models/ApiModels/Request/SharedFileCollectionMemberSaveInput.cs
using System.Text.Json.Serialization;

namespace AiTrainer.Web.Domain.Models.ApiModels.Request;


public sealed record SharedFileCollectionMemberSaveInput 
{
    public const string CollectionIdJsonName = "collectionId";
    public const string MembersToShareToJsonName = "membersToShareTo";
    [JsonPropertyName(CollectionIdJsonName)]
    public required Guid CollectionId { get; init; }
    [JsonPropertyName(MembersToShareToJsonName)]
    public required IReadOnlyCollection<SharedFileCollectionSingleMemberSaveInput> MembersToShareTo { get; init; }
}

File: src/aitrainer-test-client/src/Models/SharedFileCollectionMemberSaveInput.ts
export type SharedFileCollectionMemberSaveInput = {
  collectionId: string;
  membersToShareTo: SharedFileCollectionSingleMemberSaveInput[];
};

export type SharedFileCollectionSingleMemberSaveInput = {
  userId: string;
  canViewDocuments: boolean;
  canDownloadDocuments: boolean;
  canCreateDocuments: boolean;
  canRemoveDocuments: boolean;
};
